python/ray/util/sgd/torch/training_operator.py

```diff
@@ -33,57 +44,85 @@ def _is_multiple(component):
 
 
 class TrainingOperator:
-    """Abstract class for custom training or validation loops.
-
-    The scheduler will only be called at a batch or epoch frequency, depending
-    on the user parameter. Be sure to set ``scheduler_step_freq`` in
-    ``TorchTrainer`` to either "batch" or "epoch" to increment the scheduler
-    correctly during training. If using a learning rate scheduler
-    that depends on validation loss, you can use ``trainer.update_scheduler``.
-
-    For both training and validation, there are two granularities that
+    """Abstract class to define training state as well as custom training or
+    validation loops.
+
+    You must subclass this class and override the ``setup`` method to define
+    your training components such as the model, optimizer, data, loss,
+    and scheduler. When you pass this class to ``TorchTrainer``, a copy of
+    this class will be made on each worker.
+
+    .. code-block:: python
+
+        class MyTrainingOperator(TrainingOperator):
+
+            def setup(config):
+                model = nn.Linear(1, 1)
+                optimizer = torch.optim.SGD(
+                    model.parameters(), lr=config.get("lr", 1e-4))
+                loss = torch.nn.MSELoss()
+
+                batch_size = config["batch_size"]
+                train_data, val_data = LinearDataset(2, 5), LinearDataset(2, 5)
+                train_loader = DataLoader(train_data, batch_size=batch_size)
+                val_loader = DataLoader(val_data, batch_size=batch_size)
+
+                self.model, self.optimizer = self.register(
+                                                    models=model,
+                                                    optimizers=optimizer,
+                                                    train_loader=train_loader,
+                                                    validation_loader=val_loader,
+                                                    criterion=loss)
+        trainer = TorchTrainer(
+            training_operator_cls=MyTrainingOperator,
+            config={"batch_size": 32},
+            use_gpu=True
+        )
+        for i in range(4):
+            trainer.train()
+
+    This class provides default implementations for training and validation.
+    Make sure you set ``self.model``, ``self.optimizer``, and
+    ``self.criterion`` to leverage the default training and validation loops.
+    If ``self.scheduler`` is set, it will only be called at a batch or epoch
+    frequency, depending on the user parameter. Be sure to set
+    ``scheduler_step_freq`` in ``TorchTrainer`` to either "batch" or "epoch"
+    to increment the scheduler correctly during training. If using a
```

Comment:
```suggestion
    Set ``self.model``, ``self.optimizer``, and
    ``self.criterion`` to leverage the default training and validation loops.
    If ``self.scheduler`` is set, it will only be called at a batch or epoch
    frequency, depending on the user parameter. Set
    ``scheduler_step_freq`` in ``TorchTrainer`` to either "batch" or "epoch"
    to increment the scheduler correctly during training. If using a
```